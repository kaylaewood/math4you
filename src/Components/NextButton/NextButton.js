import React from 'react';
import './NextButton.scss';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { updateTimer } from '../../actions/updateTimer';


function NextButton({ nextLink, updateTimer }) {
  const resetInfo = () => {
    if(nextLink === '/end') {
      updateTimer('isOver', false);
    }
  }

  return (
    <Link to = {nextLink}>
      <button
        className='nextButton'
        onClick={resetInfo()}
      >let's go</button>
    </Link>
  );
}

export const mapDispatchToProps = dispatch => ({
  updateTimer: (propertyToChange, updatedValue) => dispatch(updateTimer(propertyToChange, updatedValue))
})

export default connect(null, mapDispatchToProps)(NextButton);
