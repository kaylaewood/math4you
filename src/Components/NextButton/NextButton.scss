.nextButton {
  border: none;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 25vw;
  margin-left: -25vw;
  p {
    font-size: 1.3vw;
    margin: 0;
  }
  img {
    width: 7vw;
  }
}
