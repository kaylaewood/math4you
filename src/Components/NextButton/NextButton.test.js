import React from 'react';
import ReactDOM from 'react-dom';
import { NextButton } from './NextButton';
import { shallow } from 'enzyme';

describe('NextButton', () => {
  let wrapper;
  let mockNextLink = '/';
  let mockUpdateTimer = jest.fn();
  let mockGameStats = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<NextButton
      nextLink={mockNextLink}
      updateTimer={mockUpdateTimer}
      resetGameStats={mockGameStats}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
