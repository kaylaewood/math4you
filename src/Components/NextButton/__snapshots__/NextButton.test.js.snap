// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextButton should match the snapshot 1`] = `ShallowWrapper {}`;
